coordinate-systems.md

```diff
@@ -0,0 +1,246 @@
+*   _stage_: the origin and rotation of this frame of reference are in the center of a known safe playing area like a VR room.
+*   _head-model_: this frame of reference is like eye-level except the translation component of the device pose is basically identity with a neck model. This is for experiences which expect the user to not move and only look around.
+
+These are basically different ways of expressing the transform of the device in 3D space. For example, virtual objects would have the same transform in _eye-level_, _stage_, and _head-model_ FORs, it's only the device that behaves differently (more on this later). _Stage_ basically allows for a virtual translation to be applied to the native device transform and _head-model_ always locks the device translation to (0, 0, 0).
+
+Finally, XRCoordinateSystem was not designed to be dynamic. If we want to use XRCoordinateSystems to track moving objects or reference frames, we need to design a system/API to handle that dynamicism. If we want to be able to express things like hit-test input rays relative to a device or input-source, we need to be able to represent that relationship. We also need to know what coordinate system the results are in.
+
+
+# Analysis
+
+
+## XRFrameOfReference
+
+
+### Basics
+
+The current API for using XRFrameOfReference (FOR) is as follows:
+
+
+```
+const frameOfReference = await session.requestFrameOfReference('stage');
+```
+
+
+Now that we have the frame of reference we want, we convert any XRCoordinateSystem into 'stage-space' by transforming by that FOR. Most simply, if we want the current transform of the device, we say:
+
+
+```
+let devicePose = xrFrame.getDevicePose(frameOfReference);
+```
+
+
+This is the pose of the device in the given FOR. If you want to think in terms of 'absolute space', the origin of all 3 styles of FORs is the start-of-session pose of the device - that is the device pose when the AR system was first activated. In fact, the documentation around the method XRFrameOfReference::TransformBasePose(TransformationMatrix base_pose) explains this well: "Transforms a given pose from a "base" coordinate system used by the XR service to the frame of reference's coordinate system." What this means is pretty simple - the conceit is that most transforms will naturally be expressed in some native system that has its own reference frame - in most cases it is relative to the start-of-service of the underlying AR/VR runtime.
+
+
+### Concrete Example
+
+A sphere is at position (11, 0, 0) in the native start-of-service reference frame. the app uses the stage FOR in order to support a room-scale VR experience with teleportation. Teleportation is functionality the app uses to allow the user to relocate their stage/room within the virtual world - basically to let the user teleport their room through space. The user is currently standing at the origin and has never teleported.
+
+The position of the sphere is also at (11, 0, 0) in the stage FOR.
```

Comment:
This is a great point and helps define 'stage' better for me. So basically the stage is actually something like (0, -2, 0) relative to start-of-service, meaning the sphere is at (11, 0, 0) in start-of-service but is at (11, 2, 0) in 'stage' - does that match what you are saying?

As for being at same orientation as the stage, I intentionally am trying to create a toy example here so that was a conceit - I should have brought that up as a specific simplifying assumption, however - I will add that!

Reply:
The room-scale `"stage"` will typically have been drawn by the user during room setup in the largest open space within their room, which may be well away from the desk chair where they've centered their seated-scale `"eye-level"` experiences.  I would not expect the XZ coordinates or orientation of those two frames to generally coincide.